CSharpMinifier/Rewriters/CSharpRewriter.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Rename;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace CSharpMinifier.Rewriters
+{
+    class CSharpRewriter : CSharpSyntaxRewriter
+    {
+        private SemanticModel _semanticModel;
+        private AdhocWorkspace _workspace;
+        private MinifierOptions _options;
+        private IdentifierGenerator _identifierGenerator; 
+
+        public CSharpRewriter(MinifierOptions options = null, bool visitIntoStructuredTrivia = true) : base(visitIntoStructuredTrivia)
+        {
+            _options = options ?? new MinifierOptions();
+            _identifierGenerator = new IdentifierGenerator();
+            //Add cause MSBuild does not work without it
+            var _ = typeof(Microsoft.CodeAnalysis.CSharp.Formatting.CSharpFormattingOptions);
+        }
+
+        public SyntaxNode VisitAndRename(SyntaxNode node)
```

Comment:
I think the `Visitor` functionality doesn't matter in this case. Rename method to `MinifyNames` or `MinifyNamesRecursive`.